### PR TITLE
Added a pause feature

### PR DESCRIPTION
**Pull request by johannptl**
Hi there,

I hope you're doing well! I wanted to contribute to this awesome Tetris CLI version because I've become addicted to it. I found myself playing it during boring lectures, and it's been a lifesaver. However, there was one missing feature - a pause function.

In class, when the teacher would approach, I had to pretend to be working, and I often ended up losing my game. As a result, I couldn't set any high scores. To solve this issue, I've added a simple pause function. Now, you can pause the game by pressing the 'P' key.

I hope this improvement makes the game even more enjoyable for everyone.

Happy gaming!

### CODE REVIEW

File: src/game.rs [hunk 1]
```diff
@@ -112,6 +113,12 @@ impl GameState {
     }
 
     pub fn play(&mut self, cnv: &mut Canvas, inp: &mut KeyReader, hs_disp: &Vec<&String>) -> u64 {
+
+        let pause_top    = "╔══════╗".to_string();
```
Comment (blueOkiris) [inline src/game.rs]: I would prefer it to use `draw_strs` (see line 270 for an example usage) and then move this to the top as a const like `BORDER` (and also add a space between border and text) :

```
const PAUSE: [&'static str; DISP_HEIGHT as usize] = [
    "╔════════╗",
    "║ PAUSED ║",
    "╚════════╝"
];
```

And then call it via `cnv.draw_strs(&PAUSE.to_vec(), (7, 13), SCORE_COLOR, &Reset)`

File: src/game.rs [hunk 2]
```diff
@@ -131,6 +138,16 @@ impl GameState {
                 }, UpdateEndState::Lost => {
                     break;
                 }
+                UpdateEndState::Pause => {
+                    // Keep the game paused until 'p' is pressed again
+                    loop {
```
Comment (blueOkiris) [inline src/game.rs]: Can you make this a while loop?

```
while inp.get_key() != b'p' {
    ...
}
```

Also, the actions in the pause sub-loop are pretty self-explanatory, so there's no need for any comments

File: src/main.rs [hunk 3]
```diff
@@ -40,10 +40,10 @@ const MENU: [&'static str; DISP_HEIGHT as usize] = [
     "║ - a/d - left/right ║",
     "║   - q/e - rotate   ║",
     "║ - s -> drop piece  ║",
+    "║    - p -> pause    ║",
```
Comment (blueOkiris) [inline src/main.rs:43]: Nothing wrong with this, but I wanted to note that the "p for pause" thing should be added to the README as well under the `Controls` header

**Review by blueOkiris — CHANGES_REQUESTED** [review]
Thanks for contributing! Pausing is a good feature to have.

I have a few things I'd like changed as you can see below, but also, can you bump the version number in Cargo.toml/.lock to `23.96.1`?

Thanks!

**Comment by johannptl** [comment]
Thanks for the advice. I changed everything now. Sorry for my unclean code.

File: src/main.rs [hunk 4]
```diff
@@ -52,7 +52,7 @@ const MENU: [&'static str; DISP_HEIGHT as usize] = [
     "║                    ║",
     "║                    ║",
     "╚════════════════════╝",
-    "                      "
+    "                      ",
```
Comment (blueOkiris) [inline src/main.rs]: Can you remove this extra comma here?

**Review by blueOkiris — CHANGES_REQUESTED** [review]
Sorry. One last little thing.

**Comment by johannptl** [comment]
I did. Sorry for that one, my bad.

**Comment by blueOkiris** [comment]
Alright everything looks good I think. Thanks for the help!

**Review by blueOkiris — APPROVED** [review]

**Pull request merged**